### PR TITLE
ACCS-1605 - Add support for FeatureGroups and Setups

### PR DESCRIPTION
**Pull request by lukeedw**

### CODE REVIEW

File: Delcam.ProductInterface.PowerMILL/Classes/PMFeatureGroupsCollection.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        #region Operations
+
+        /// <summary>
+        /// Initialises the collection with the items in PowerMILL.
+        /// </summary>
+        internal void Initialise()
+        {
+            if (_powerMILL.Version.Major >= 2018)
+            {
+                foreach (string name in ReadFeatureGroups())
```
Comment (danInCode) [inline Delcam.ProductInterface.PowerMILL/Classes/PMFeatureGroupsCollection.cs:42]: This is not going to execute in every foreach loop?

File: Delcam.ProductInterface.PowerMILL/Classes/PMSetupsCollection.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        #region Operations
+
+        /// <summary>
+        /// Initialises the collection with the items in PowerMILL.
+        /// </summary>
+        internal void Initialise()
+        {
+            if (_powerMILL.Version.Major >= 2019)
+            {
+                foreach (string name in ReadSetups())
```
Comment (danInCode) [inline Delcam.ProductInterface.PowerMILL/Classes/PMSetupsCollection.cs:42]: The same in here?

**Review by danInCode — CHANGES_REQUESTED** [review]

File: Delcam.ProductInterface.PowerMILL/Classes/PMFeatureGroupsCollection.cs (hunk 1 above)
Comment (lukeedw) [reply Delcam.ProductInterface.PowerMILL/Classes/PMFeatureGroupsCollection.cs:42]: No, that's not how they work.  The iteration function is only called once.
https://stackoverflow.com/questions/2447559/c-sharp-does-function-get-called-for-each-iteration-of-a-foreach-loop
I tested it to be doubly sure!

File: Delcam.ProductInterface.PowerMILL/Classes/PMSetupsCollection.cs (hunk 2 above)
Comment (lukeedw) [reply Delcam.ProductInterface.PowerMILL/Classes/PMSetupsCollection.cs:42]: See other reply.

**Review by danInCode — APPROVED** [review]

**Pull request merged**